android/src/main/java/com/brentvatne/common/react/VideoEventEmitter.kt

```diff
@@ -0,0 +1,349 @@
+package com.brentvatne.common.react
+
+import com.brentvatne.common.api.TimedMetadata
+import com.brentvatne.common.api.Track
+import com.brentvatne.common.api.VideoTrack
+import com.brentvatne.exoplayer.ReactExoplayerView
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.bridge.WritableArray
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.uimanager.ThemedReactContext
+import com.facebook.react.uimanager.UIManagerHelper
+import com.facebook.react.uimanager.events.Event
+import com.facebook.react.uimanager.events.EventDispatcher
+import java.io.PrintWriter
+import java.io.StringWriter
+
+enum class EventTypes(val eventName: String) {
+    EVENT_LOAD_START("onVideoLoadStart"),
+    EVENT_LOAD("onVideoLoad"),
+    EVENT_ERROR("onVideoError"),
+    EVENT_PROGRESS("onVideoProgress"),
+    EVENT_BANDWIDTH("onVideoBandwidthUpdate"),
+    EVENT_CONTROLS_VISIBILITY_CHANGE("onControlsVisibilityChange"),
+    EVENT_SEEK("onVideoSeek"),
+    EVENT_END("onVideoEnd"),
+    EVENT_FULLSCREEN_WILL_PRESENT("onVideoFullscreenPlayerWillPresent"),
+    EVENT_FULLSCREEN_DID_PRESENT("onVideoFullscreenPlayerDidPresent"),
+    EVENT_FULLSCREEN_WILL_DISMISS("onVideoFullscreenPlayerWillDismiss"),
+    EVENT_FULLSCREEN_DID_DISMISS("onVideoFullscreenPlayerDidDismiss"),
+
+    EVENT_READY("onReadyForDisplay"),
+    EVENT_BUFFER("onVideoBuffer"),
+    EVENT_PLAYBACK_STATE_CHANGED("onVideoPlaybackStateChanged"),
+    EVENT_IDLE("onVideoIdle"),
+    EVENT_TIMED_METADATA("onTimedMetadata"),
+    EVENT_AUDIO_BECOMING_NOISY("onVideoAudioBecomingNoisy"),
+    EVENT_AUDIO_FOCUS_CHANGE("onAudioFocusChanged"),
+    EVENT_PLAYBACK_RATE_CHANGE("onPlaybackRateChange"),
+    EVENT_VOLUME_CHANGE("onVolumeChange"),
+    EVENT_AUDIO_TRACKS("onAudioTracks"),
+    EVENT_TEXT_TRACKS("onTextTracks"),
+
+    EVENT_TEXT_TRACK_DATA_CHANGED("onTextTrackDataChanged"),
+    EVENT_VIDEO_TRACKS("onVideoTracks"),
+    EVENT_ON_RECEIVE_AD_EVENT("onReceiveAdEvent");
+
+    companion object {
+        fun toMap() =
+            mutableMapOf<String, Any>().apply {
+                EventTypes.entries.forEach { eventType ->
+                    put("top${eventType.eventName.removePrefix("on")}", mapOf("registrationName" to eventType.eventName))
+                }
+            }
```

Comment:
Will this work also with "old architecture"? If not we should add "if" here to don't do it on "old arch"

Reply:
It works properly on old arch

Reply:
Thanks, then let's merge after CI is done